File: src/test/java/de/qabel/core/crypto/QblEcPairTestFactory.java
package de.qabel.core.crypto;

import org.meanbean.lang.Factory;

/**
 * QblPrimaryKeyPairTestFactory
 * Creates distinct instances of class QblPrimaryKeyPair
 * Attention: For testing purposes only
 */
public class QblEcPairTestFactory implements Factory<QblECKeyPair> {
    QblECKeyPair ecKeyPair;

    public QblEcPairTestFactory() {
    }

    @Override
    public QblECKeyPair create() {
        ecKeyPair = new QblECKeyPair();
        return ecKeyPair;
    }
}

File: core/src/test/java/de/qabel/core/crypto/QblEcPairTestFactory.kt
package de.qabel.core.crypto

import de.qabel.core.crypto.QblECKeyPair
import org.meanbean.lang.Factory

/**
 * QblPrimaryKeyPairTestFactory
 * Creates distinct instances of class QblPrimaryKeyPair
 * Attention: For testing purposes only
 */
class QblEcPairTestFactory : Factory<QblECKeyPair> {
    lateinit internal var ecKeyPair: QblECKeyPair

    override fun create(): QblECKeyPair {
        ecKeyPair = QblECKeyPair()
        return ecKeyPair
    }
}
